src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/NonInteractive/WsTrustResponse.cs

```diff
@@ -119,29 +127,31 @@ internal static WsTrustResponse CreateFromResponseDocument(XDocument responseDoc
                     }
                 }
 
-                if (parseResponse)
+                if (!parseResponse)
+                {
+                    throw new AdalException(AdalError.ParsingWsTrustResponseFailed);
+                }
+
+                IEnumerable<XElement> tokenResponses =
+                    responseDocument.Descendants(t + "RequestSecurityTokenResponse");
+                foreach (var tokenResponse in tokenResponses)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)